extras/src/main/scala/spire/math/extras/interval/IntervalSeq.scala

```diff
@@ -0,0 +1,687 @@
+package spire.math.extras.interval
+
+import java.util.Arrays
+
+import spire.algebra.{Bool, Eq, Order}
+import spire.math.interval._
+import spire.math.{Interval, Rational}
+
+import scala.annotation.tailrec
+import scala.language.implicitConversions
+import scala.reflect.ClassTag
+
+final class IntervalSeq[T] private (
+    val belowAll: Boolean,
+    private val values: Array[T],
+    private val kinds: Array[Byte],
+    private implicit val order: Order[T]) extends IntervalSet[T, IntervalSeq[T]] { lhs =>
+
+  import IntervalSeq._
+
+  private def binarySearch(key: T): Int = {
+    var low = 0
+    var high = values.length - 1
+    while (low <= high) {
+      val mid = (low + high) >>> 1
+      val midVal = values(mid)
+      val c = order.compare(midVal, key)
+      if (c < 0) {
+        low = mid + 1
+      }
+      else if (c > 0) {
+        high = mid - 1
+      }
+      else {
+        // scalastyle:off return
+        return mid
+        // scalastyle:on return
+      }
+    }
+    -(low + 1)
+  }
+
+  def below(index: Int): Boolean = {
```

Comment:
Given that this is the only version that takes an `Int`, is it worth making it (package) private?